win32/gui.py

```diff
@@ -713,15 +715,67 @@ def check_yaml_syntax(content):
         warning_popup("Unable to parse yaml: \n %s" % str(e))
         raise
 
+def restartServiceWithDeps(serviceName):
+    # stop the service recursively, looking for dependent services.
+    # if it fails, attempt to restart the ones that were stopped
+    # (so that things are left in the correct state)
+    stopped_deps = list()
+    hscm = win32service.OpenSCManager(None,None,win32service.SC_MANAGER_ALL_ACCESS)
+    try:
+        deps = win32serviceutil.__FindSvcDeps(serviceName)
+
+        for dep in deps:
+            hs = win32service.OpenService(hscm, dep, win32service.SERVICE_ALL_ACCESS)
+            try:
+                win32serviceutil.__StopServiceWithTimeout(hs)
+                stopped_deps.insert(0, dep)
+            except pywintypes.error as exc:
+                if exc.winerror != winerror.ERROR_SERVICE_NOT_ACTIVE:
+                    # walk through and start all the services that have been stopped
+                    for restart_dep in stopped_deps:
+                        try:
+                            win32serviceutil.StartService(restart_dep)
+                        except:
```

Comment:
nit: can you change all the `except:` to `except Exception:`? Always better to avoid the catch-all 👍